community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NodeRelationshipCache.java

```diff
@@ -410,7 +412,7 @@ public int getCount( long nodeId, Direction direction )
         {
             this.base = base;
             assert chunkSize > 0;
-            this.array = arrayFactory.newDynamicByteArray( chunkSize,
+            this.array = arrayFactory.newDynamicByteArray( 1_000_000,
```

Comment:
Perhaps I'm mistaken, but I don't think we can easily calculate this and if we use the worst-case figure many imports will allocate lots of memory it doesn't need.